api/app/wildfire_one.py

```diff
@@ -642,9 +642,17 @@ def parse_hourly_actual(station_code: int, hourly_reading: WeatherReading):
                        hourly_reading.datetime.strftime("%b %d %Y %H:%M:%S"),
                        hourly_reading.observation_valid_comment)
 
-    is_valid = temp_valid and rh_valid and wdir_valid and wspeed_valid and precip_valid and is_valid_wfwx
+    is_obs_invalid = not temp_valid and not rh_valid and not wdir_valid\
+        and not wspeed_valid and not precip_valid
 
-    return None if (is_valid is False) else HourlyActual(
+    if is_obs_invalid:
+        logger.error("Hourly actual not written to DB for station code %s at time %s: %s",
+                     station_code, hourly_reading.datetime.strftime("%b %d %Y %H:%M:%S"),
+                     hourly_reading.observation_valid_comment)
+
+    # don't write the HourlyActual to our database if every value is invalid. If even one
+    # weather variable observed is valid, write the HourlyActual to DB.
+    return None if is_obs_invalid else HourlyActual(
```

Comment:
Add checks for None, replacing with Math.nan (and please explain why - consistency with forecasts)